pkg/coscheduling/coscheduling.go

```diff
@@ -132,11 +133,45 @@ func (cs *Coscheduling) Less(podInfo1, podInfo2 *framework.QueuedPodInfo) bool {
 func (cs *Coscheduling) PreFilter(ctx context.Context, state *framework.CycleState, pod *v1.Pod) *framework.Status {
 	if err := cs.pgMgr.PreFilter(ctx, pod); err != nil {
 		klog.Error(err)
+		state.Write(cs.getStateKey(), NewNilStateData())
 		return framework.NewStatus(framework.Unschedulable, err.Error())
 	}
 	return framework.NewStatus(framework.Success, "")
 }
 
+// PostFilter is use for rejecting the group pods if a pod does not pass PreFilter or Filter.
+func (cs *Coscheduling) PostFilter(ctx context.Context, state *framework.CycleState, pod *v1.Pod,
+	filteredNodeStatusMap framework.NodeToStatusMap) (*framework.PostFilterResult, *framework.Status) {
+	// check if failed when pre filter
+	_, err := state.Read(cs.getStateKey())
+	if err == nil {
+		state.Delete(cs.getStateKey())
+		return &framework.PostFilterResult{}, framework.NewStatus(framework.Unschedulable)
+	}
+
+	pgName, pg := cs.pgMgr.GetPodGroup(pod)
+	if pg == nil {
+		klog.V(4).Info("Pod does not belong to any group")
+		return &framework.PostFilterResult{}, framework.NewStatus(framework.Unschedulable, "can not find pod group")
+	}
+
+	assigned := cs.pgMgr.CalculateAssignedPods(pg.Name, pod.Namespace)
+	if assigned >= int(pg.Spec.MinMember) {
+		klog.Errorf("assigned %v", assigned)
+		return &framework.PostFilterResult{}, framework.NewStatus(framework.Unschedulable)
+	}
+
+	cs.frameworkHandler.IterateOverWaitingPods(func(waitingPod framework.WaitingPod) {
```

Comment:
The updated logic is a bit conservative now.

However, I think it's based on an implicit assumption: if the n<sub>th</sub> Pod failed, it's inferrable other Pods belonging to the same PodGroup would be very likely to fail. Am I understanding it right?

If so, we should comment it so as not to confuse readers.

Reply:
> However, I think it's based on an implicit assumption: if the nth Pod failed, it's inferrable other Pods belonging to the same PodGroup would be very likely to fail. Am I understanding it right?

Yes, will add comments